src/NServiceBus.Metrics.ServiceControl/ServiceControlReporting/MetricsContext.cs

```diff
@@ -0,0 +1,100 @@
+﻿namespace NServiceBus.Metrics.ServiceControl.ServiceControlReporting
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+
+    class MetricsContext
+    {
+        public MetricsContext(string context)
+        {
+            Context = context;
+        }
+
+        readonly ConcurrentDictionary<string, Counter> counters = new ConcurrentDictionary<string, Counter>();
+        readonly ConcurrentDictionary<Tuple<string,string>, Gauge> gauges = new ConcurrentDictionary<Tuple<string, string>, Gauge>();
+
+        public IEnumerable<Counter> Counters => counters.Select(kvp => kvp.Value);
+        public IEnumerable<Gauge> Gauges => gauges.Select(kvp => kvp.Value);
+        public string Context { get; }
+
+        public Counter Counter(string key) => counters.GetOrAdd(key, k => new Counter(k));
+        public Gauge Gauge(string key, string queue) => gauges.GetOrAdd(Tuple.Create(key,queue), t => new Gauge(t.Item1,t.Item2));
+
+        public string ToJson()
+        {
+            return SimpleJson.SerializeObject(this);
+        }
+    }
+
+    class Gauge
+    {
+        long value;
+
+        public Gauge(string key, string queue)
+        {
+            Tags = new[]
+            {
+                $"key:{key}",
+                $"queue:{queue}",
+                "type:queue-length.received"
+            };
+            Name = $"Received sequence for {key}";
+        }
+
+        public void Report(long v)
+        {
+            Volatile.Write(ref value, v);
+        }
+
+        public double Value => Volatile.Read(ref value);
+        public string Name { get; }
+        public string[] Tags { get; }
+    }
+
+    class Counter
+    {
+        long value;
+
+        public Counter(string key)
+        {
+            Tags = new[]
+            {
+                "type:queue-length.sent",
+                $"key:{key}"
+            };
+            Name = $"Sent sequence for {key}";
```

Comment:
I don't think we need this part. SC.Monitoring is not using `Name` in any way.

Reply:
This was used previously for recognizing right gauges/counters in tests and still is. Should I remove it?